frontend/app/models/file-entry.js

```diff
@@ -10,12 +10,24 @@ export default class FileEntry extends Model {
     if (this.isDeleted) {
       return super.save();
     }
+
     let url = `/api/accounts/${this.account.get("id")}/file_entries`;
     let opts = {
       data: { description: this.description },
       headers: { "X-CSRF-Token": this.csrfToken }
     };
 
-    return this.file.upload(url, opts);
+    let promise = this.file.upload(url, opts);
+    promise
+      .then( savedRecords => {
+        let data = JSON.parse(savedRecords.body).data
+        this.id = data.id;
+        this.filename = data.attributes.filename;
+      })
+      .catch(() => {
+        this.account = null;
+      });
```

Comment:
Didn't work in `handleSubmitError` in [file-entry.js](https://github.com/puzzle/cryptopus/blob/55a70697d667a38f3e7029c4b22678d6e5fa63ab/frontend/app/components/file-entry-form.js#L62).

Reply:
hmm can you not directly set the account on the record inside file-entry-form? Not using the changeset
`this.record.account = null`

Reply:
Thanks for this suggestion, works.